src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,763 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+            }
+
+            // For One - Many and Many - Many relationship types, the entire logic needs to be run for each element of the input.
+            // So, when the input parameters is of list type, we iterate over the list and run the logic for each element.
+            if (multipleCreateStructure.InputMutParams.GetType().GetGenericTypeDefinition() == typeof(List<>))
+            {
+                List<IDictionary<string, object?>> parsedInputParamList = (List<IDictionary<string, object?>>)multipleCreateStructure.InputMutParams;
+                List<IValueNode> paramList = (List<IValueNode>)parameters;
+                int idx = 0;
+
+                foreach (IDictionary<string, object?> parsedInputParam in parsedInputParamList)
+                {
+                    MultipleCreateStructure multipleCreateStrucutreForCurrentItem = new(entityName: multipleCreateStructure.EntityName,
+                                                                                        higherLevelEntityName: multipleCreateStructure.HigherLevelEntityName,
+                                                                                        inputMutParams: parsedInputParam,
+                                                                                        isLinkingTableInsertionRequired: multipleCreateStructure.IsLinkingTableInsertionRequired)
+                    {
+                        CurrentEntityParams = multipleCreateStructure.CurrentEntityParams,
+                        LinkingTableParams = multipleCreateStructure.LinkingTableParams
+                    };
+
+                    Dictionary<string, Dictionary<string, object?>> primaryKeysOfCreatedItems = new();
+                    IValueNode? nodeForCurrentInput = paramList[idx];
+                    if (nodeForCurrentInput is null)
+                    {
+                        throw new DataApiBuilderException(message: "Error when processing the mutation request",
+                                                          statusCode: HttpStatusCode.InternalServerError,
+                                                          subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+                    }
+
+                    PerformDbInsertOperation(context, nodeForCurrentInput.Value, sqlMetadataProvider, multipleCreateStrucutreForCurrentItem, nestingLevel);
+                    idx++;
+                }
+            }
+            else
+            {
+                if (parameters is not List<ObjectFieldNode> parameterNodes)
+                {
+                    throw new DataApiBuilderException(message: "Error occurred while processing the mutation request",
+                                                      statusCode: HttpStatusCode.InternalServerError,
+                                                      subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+                }
```

Comment:
I recognize that you mentioned hotchocolate would handle circumstances where parameters is not List<ObjectFieldNode>, but we also protect against that here. Regardless, doesn't hitting this condition indicate that the user sumbmitted a malformed request -> HTTP 400

Reply:
Right. It should be bad request.